src/init/agent.rs

```diff
@@ -212,11 +214,20 @@ impl Agent {
 
         let mut pipeline_flush_sub = self.pipeline_flush_sub.take();
 
+        // AWS-XRay only supports a batch size of 50 segments
+        let mut trace_batch_config = build_traces_batch_config(config.batch.clone());
+        if config.exporter == Exporter::AwsXray && trace_batch_config.max_size > 50 {
+            info!(
+                "AWS X-Ray only supports a batch size of 50 segments, setting batch max size to 50"
+            );
+            trace_batch_config.max_size = 50;
```

Comment:
We may need to revisit how we handle the continual batch splitting, but I was also hitting that on some other testing.

Reply:
Yeah we have to tackle that soon. It should be too much work